src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1488,19 +1706,36 @@ export function onRequest (context, next) {
 };
 ```
 
-### `next()`
+#### `context`
+
+<p>
+
+**Type:** `APIContext`
+</p>
+
+The first argument of `onRequest()` is a context object. It mirrors many of the `Astro` global properties.
+
+<ReadMore>See [Endpoint contexts](#endpoint-context) for more information about the context object.</ReadMore>
+
+#### `next()`
+
+<p>
+
+**Type:** `(rewritePayload?: string | URL | Request) => Promise<Response>`<br />
+</p>
 
 A function that intercepts (reads and modifies) the `Response` of a `Request` or calls the "next" middleware in the chain and returns a `Response`. For example, this function could modify the HTML body of a response.
 
 This is an optional argument of `onRequest()`, and may provide the required `Response` returned by the middleware.
```

Comment:
Not changed in this PR, but taking advantage to suggest some changes as I don’t think this is particularly clear or accurate. This proposal also aligns  the phrasing a bit with the new `context` docs.

```suggestion
The second argument of `onRequest()` is a function that calls all the subsequent middleware in the chain and returns a `Response`. For example, other middleware could modify the HTML body of a response and awaiting the result of `next()` would allow your middleware to respond to those changes.
```